File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int height;
    void pyramidBricks(int i);

    // Prompt user for the height of the pyramid
    do
    {
        height = get_int("Height: ");
    }
    while (height > 8 || height < 1);

    // Print the pyramid
    for (int i = 0; i < height; i++)
    {
        // Print spaces
        for (int x = height - i; x > 1; x--)
        {
            printf(" ");
        }
        // Print left side of pyramid
        pyramidBricks(i);

        //Print middle gap
        printf("  ");

        // Print right side of pyramid
        pyramidBricks(i);
        printf("\n");
    }
}

// Function to print pyramid bricks
void pyramidBricks(int i)
{
    for (int j = 0; j <= i; j++)
    {
        printf("#");
    }
}

File: mario.py
import cs50


def main():
    # Initialise height variable for user input
    height = 0
    row = 0
    # Set height to user input for the desired pyramid height
    while height < 1 or height > 8:
        height = cs50.get_int("Height: ")

    pyramid(height, row)

# Recursively print pyramid to console


def pyramid(height, row):
    if height > 1:
        pyramid(height - 1, row + 1)
    print(" " * row, end="")
    print("#" * height)


main()